src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -486,6 +512,7 @@ fn maybe_lib<'a>(
 /// build script.
 fn dep_build_script<'a>(
     unit: &Unit<'a>,
+    dep_kind: DepKind,
```

Comment:
Similar to above, running the build script needs to set the `CARGO_FEATURE_*` environment variables for its parent package.  It needs to know if the parent is being built as a build-dependency or not so it can set those env vars correctly.